cyipopt/tests/unit/test_scipy_optional.py

```diff
@@ -33,9 +33,11 @@ def test_minimize_ipopt_import_error_if_no_scipy():
 @pytest.mark.skipif("scipy" not in sys.modules,
                     reason="Test only valid if Scipy available.")
 def test_minimize_ipopt_input_validation():
+    from scipy import optimize
+
     x0 = 1
     def f(x):
-        return x**2
+        return x @ x
```

Comment:
Is this changing functionality?

```
In [3]: import numpy as np

In [4]: x = np.array([1])

In [5]: x @ x
Out[5]: 1

In [6]: type(x @ x)
Out[6]: numpy.int64

In [7]: type(x**2)
Out[7]: numpy.ndarray
```

Reply:
I think I changed this line just to generalize it to be a generic multivariate quadratic bowl rather than only working for scalars. It still works for scalars, though.

```
from cyipopt import minimize_ipopt
def f(x):
    return x ** 2
minimize_ipopt(f, x0=1)
```
returning
```
     fun: 0.0
    info: {'x': array([0.]), 'g': array([], dtype=float64), 'obj_val': 0.0, 'mult_g': array([], dtype=float64), 'mult_x_L': array([0.]), 'mult_x_U': array([0.]), 'status': 0, 'status_msg': b'Algorithm terminated successfully at a locally optimal point, satisfying the convergence tolerances (can be specified by options).'}
 message: b'Algorithm terminated successfully at a locally optimal point, satisfying the convergence tolerances (can be specified by options).'
    nfev: 7
     nit: 1
    njev: 3
  status: 0
 success: True
       x: array([0.])
```